pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -87,7 +87,51 @@ let
       args = rattrs (args // { inherit finalPackage overrideAttrs; });
       #              ^^^^
 
-      overrideAttrs = f0: makeDerivationExtensible (lib.extends (lib.toExtension f0) rattrs);
+      overrideAttrs =
+        f0:
+        let
+          extends' =
+            overlay: f:
+            (
+              final:
+              let
+                prev = f final;
+                thisOverlay = overlay final prev;
+                # warnForBadVersionOverride = (thisOverlay ? version && !(thisOverlay ? src));
+                warnForBadVersionOverride = (
+                  thisOverlay ? version
+                  && !(thisOverlay ? src)
+                  && args.src ? outputHash
```

Comment:
This is here so that the warning only happens when `src` is an FOD, right? Why is the warning not applicable when `src` is not an FOD?

Reply:
Right.

So the logic for that was if `src` isn't a FOD, changes in `version` likely aren't going to affect `src` -- and if they do, it'll be done properly, e.g. `src = ./foo-${version}`.

Wait, hm. You're right, that would realistically only be the ideal case with `finalAttrs`, if it was `rec` then you'd still want to override just `src`. I'll change it.

Reply:
Yea that does make sense. The thing it made me think of is this _incredibly strange_ thing we do with edk2: https://github.com/NixOS/nixpkgs/blob/49ee76a57b4392fc51c34829bbb0f9ed944bbb30/pkgs/by-name/ed/edk2/package.nix#L35-L49

But more generally, scenarios where `src` is a wrapper around an FOD that adds some kind of processing. (In edk2's case, I think it could probably be a `patchPhase` thing but that's beside the point).

Reply:
Oh god that's cursed.

So I guess it'd be best to remove the FOD check, right?

Reply:
I *think* so, but it won't bother me either way tbh.